### PR TITLE
Fix bug in InputNormalization

### PR DESCRIPTION
**Pull request by goexle**
change global mean and std only if the model is in training mode. Do not change it in eval mode.

### CODE REVIEW

**Comment by mravanelli** [comment]
Thank you for reporting that. I though this was already solved in the past. Let me investigate a bit and go back to you.

**Comment by mravanelli** [comment]
I took a look at it and we fixed it in #933 (a big PR that will be part of the next major version of speechbrain). I did some tests on TIMIT and LibriSpeech and this, as expected, has no effect on the performance. The count variable is very high after training and the test statistics has a weight that is close to zero.  We can, however, merge this PR to fix that even in the current version.

**Comment by goexle** [comment]
@mravanelli  thank you for reviewing and merging

**Comment by danpovey** [comment]
I am a bit skeptical about how easy it would be to apply this kind of normalization in a product.
In our setup we recently found that simply multiplying the input log-energy features by 0.1 works a little better
than applying feature batch-normalization.  This allows the bias parameter of the 1st layer to be effective
(since optimizers won't usually allow it to reach a large value, like 10, to be comparable to the magnitude of
the input features).

**Comment by mravanelli** [comment]
That's pretty interesting!

**Comment by danpovey** [comment]
Yeah we also found we can remove the batchnorm from the conformer
convolution without degrading the results.
